File: 228-summary-ranges/summary-ranges.cs
public class Solution {
    public IList<string> SummaryRanges(int[] nums) {
        List<string> ranges = new List<string>();
        
        if (nums.Length == 0) return ranges;

        int start = nums[0];
        int end = nums[0];

        for (int i = 1; i < nums.Length; i++) {
            if (nums[i] == end + 1) {
                end = nums[i];
            } else {
                if (start == end) {
                    ranges.Add(start.ToString());
                } else {
                    ranges.Add($"{start}->{end}");
                }
                start = nums[i];
                end = nums[i];
            }
        }

        if (start == end) {
            ranges.Add(start.ToString());
        } else {
            ranges.Add($"{start}->{end}");
        }

        return ranges;
    }
}

File: 228-summary-ranges/summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    const ranges = [];
    if (nums.length === 0) return ranges;
    
    let start = nums[0];
    
    for (let i = 1; i <= nums.length; i++) {
        if (i === nums.length || nums[i] !== nums[i - 1] + 1) {
            if (start === nums[i - 1]) {ranges.push(`${start}`);} 
            else {ranges.push(`${start}->${nums[i - 1]}`);}
            start = nums[i];
        }
    }
    
    return ranges;
};
